compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -200,19 +201,19 @@ impl<'a, 'tcx> SpanEncoder for EncodeContext<'a, 'tcx> {
         }
     }
 
-    fn encode_symbol(&mut self, symbol: Symbol) {
-        // if symbol predefined, emit tag and symbol index
-        if symbol.is_predefined() {
+    fn encode_symbol_index_or_byte_str(&mut self, index: u32, byte_str: &[u8]) {
```

Comment:
```suggestion
    fn encode_symbol(&mut self, index: u32) {
```
`byte_str` can be obtained here with `ByteSymbol::new(index).as_byte_str()`.
Can also avoid touching the interner in `SYMBOL_PREDEFINED` cases if doing it here.

Reply:
I mean the `byte_str` parameter doesn't need to be passed because it can be obtained from the `index` parameter inside this function.

Reply:
Ah, this is no longer relevant after the serialization changes.